src/common/engine.js

```diff
@@ -697,7 +729,39 @@ class CurrencyEngine {
     return dataDump
   }
 
-  // asynchronous
+  makeSpend (edgeSpendInfo: EdgeSpendInfo): Object {
+    const valid = validateObject(edgeSpendInfo, MakeSpendSchema)
+
+    if (!valid) {
+      throw new Error('Error: Invalid EdgeSpendInfo')
+    }
+
+    for (const st of edgeSpendInfo.spendTargets) {
+      if (st.publicAddress === this.walletLocalData.publicKey) {
+        throw new error.SpendToSelfError()
+      }
+    }
+
+    let currencyCode: string = ''
+    if (typeof edgeSpendInfo.currencyCode === 'string') {
+      currencyCode = edgeSpendInfo.currencyCode
+    } else {
+      currencyCode = this.currencyInfo.currencyCode
+    }
+    const nativeBalance = this.walletLocalData.totalBalances[currencyCode]
+    if (!nativeBalance || bns.eq(nativeBalance, '0')) {
+      throw new error.InsufficientFundsError()
+    }
+
+    edgeSpendInfo.currencyCode = currencyCode
```

Comment:
Bold move, mutating an object that doesn't belong to you. I think it's OK with the way the core and bridge work, though, so no need to fix it. If it were another data structure I might worry more about it.